### PR TITLE
fix(import): stop multi-season folder imports from deleting source files

### PR DESCRIPTION
**Pull request by Jsakkos**
## Summary

A user imported a multi-season folder (`X:\media\rips\Seinfeld`) and many episodes were **permanently deleted**. Investigation (DB + `engram.log`) traced two compounding, import-specific defects. Both are fixed here, test-first.

The root cause is that **an import's `staging_path` IS the user's real source folder** (`create_job_from_staging` points the job straight at it — no copy step), unlike a disc rip whose staging is a throwaway temp dir.

## Defects fixed

**1. Staging cleanup deleted the import source (data loss).**
`CleanupService.delete_staging` did `shutil.rmtree(job.staging_path)` on a terminal job under the `on_success`/`on_completion` policy, with no rip-vs-import distinction. For an import that `rmtree`'d the user's own folder — bypassing the Recycle Bin — including any content the importer never scanned. Log smoking gun: `Cleaned up staging directory: X:\media\rips\Seinfeld`.
→ Skip cleanup when `job.drive_id == "import"` (the marker `_on_staging_event` already sets for watch-folder imports). Successfully imported files are *moved* out, so there is nothing legitimate to reclaim.

**2. Flat scan ignored Season subfolders (incomplete import).**
`StagingWatcher._scan_import_dir` early-returned a single season-less `flat` unit on the first loose top-level `.mkv` it saw (`os.scandir` order is arbitrary). So a root holding both loose files **and** `Season NN/` subfolders imported only the loose files; the seasons were never scanned — then deleted by defect 1.
→ Defer the flat decision: only emit a `flat` root unit when loose MKVs exist **and** no structured subfolder (Season/disc) units were found. When both exist the subfolders win and the ambiguous loose files are left in place (logged, so the user knows to move them into a Season folder).

Together: an import can no longer delete your source, **and** a mixed loose-files + season-subfolders layout now imports the seasons. Defect 1 is defense-in-depth — even if a future scanner edge case leaves content un-imported, it can never be deleted.

## Tests (RED→GREEN)

- `TestImportSourceProtection` — import source preserved; disc-rip staging still cleaned (`test_staging_cleanup.py`).
- `test_loose_root_files_do_not_shadow_season_subdirs` — loose root files no longer shadow `Season NN/` subdirs (`test_staging_watcher.py`).

Each was watched failing for the real production reason before the fix. Full backend unit suite (1663 tests) passes; ruff + format clean.

## Not included (follow-up)

Season-less imports (`detected_season=None`) still brute-force every cached season (the tracked unknown-season runaway), match nothing, and auto-sort to `Extras` instead of routing to the Review/Inspector queue — the "Inspector is broken for multi-season imports" symptom. Deliberately deferred to keep this PR focused on the data-loss path.

## Reviewer notes

- Guard keys on `drive_id == "import"`; staging-area drops (`drive_id == "staging"`) are unchanged and still cleaned, since they live in the managed staging root.
- No schema/API changes. Behavior change is limited to import scanning + import-job cleanup.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backend/app/services/cleanup_service.py [hunk 1]
```diff
@@ -39,12 +39,32 @@ async def on_job_terminal(self, job_id: int, state: JobState) -> None:
...
+        remains, including content the importer never scanned (e.g. Season
+        subfolders skipped by the flat-file scan). Never delete an import
+        source.
+        """
         async with async_session() as session:
             job = await session.get(DiscJob, job_id)
             if not job or not job.staging_path:
                 return
 
+            if job.drive_id == "import":
```
[the remark this answers is not in the document]
Comment (Jsakkos) [reply backend/app/services/cleanup_service.py:59]: Documented the invariant in run_timed_cleanup: it only rmtrees `job_*`-prefixed dirs inside the managed staging_root, and an import source is neither under staging_root nor named `job_*`, so it's structurally unreachable — with a note to add a `drive_id == "import"` DB cross-check if that convention ever changes. Chose the comment over a DB lookup since the latter would add async DB access to the hourly loop for a path that can't hit an import. (4279162)

File: backend/app/core/staging_watcher.py [hunk 2]
```diff
@@ -296,6 +288,37 @@ def _scan_import_dir(self, root: Path) -> list[tuple[Path, int, int, dict]]:
...
+                        "structure": "flat",
+                        "show_name": None,
+                        "season": None,
+                        "destination_mode": self._import_destination_mode,
+                        "source": "import",
+                    },
+                )
+            )
+        elif root_has_loose_mkv and units:
+            loose_count, _ = self._count_mkvs(root)
```
[the remark this answers is not in the document]
Comment (Jsakkos) [reply backend/app/core/staging_watcher.py]: Fixed — loose-MKV count and size are now tallied during the single os.scandir pass and reused for both the flat unit and the log message, removing the second _count_mkvs(root) traversal and the TOCTOU window. (4279162)
